apps/wallet-mobile/src/wallet-manager/walletManager.ts

```diff
@@ -9,25 +10,105 @@ import {makeWalletEncryptedStorage} from '../yoroi-wallets/storage'
-import {AddressMode, WalletManagerEvent, WalletManagerSubscription, WalletMeta} from './types'
+import {AddressMode, WalletInfos, WalletManagerEvent, WalletManagerSubscription, WalletMeta} from './types'
 import {isWalletMeta, parseWalletMeta} from './validators'
 
+const thirtyFiveSeconds = 35 * 1e3
+
 export class WalletManager {
+  readonly #walletsRootStorage: App.Storage
+  readonly #rootStorage: App.Storage
+  readonly #openedWallets: Map<string, YoroiWallet> = new Map()
+  public readonly walletInfos$ = new Subject<WalletInfos>()
+  readonly #walletInfos: WalletInfos = new Map()
+  #selectedWalletId: YoroiWallet['id'] | null = null
+
   private subscriptions: Array<WalletManagerSubscription> = []
-  walletsRootStorage: App.Storage
-  rootStorage: App.Storage
+  private isSyncing = false
 
   constructor() {
-    this.walletsRootStorage = rootStorage.join('wallet/')
-    this.rootStorage = rootStorage
+    this.#walletsRootStorage = rootStorage.join('wallet/')
+    this.#rootStorage = rootStorage
+  }
+
+  set selectedWalletId(id: YoroiWallet['id']) {
+    this.#selectedWalletId = id
+    this._notify({type: 'selected-wallet-id', id})
+  }
+
+  get selectedWalledId() {
+    return this.#selectedWalletId
+  }
+
+  startSyncingAllWallets() {
+    const syncWallets = () => {
+      if (this.isSyncing) return
+
+      this.isSyncing = true
+
+      from(this.openWallets())
+        .pipe(
+          concatMap((wallets) => {
+            this.#walletInfos.clear()
+            wallets.forEach((wallet) => {
+              this.#walletInfos.set(wallet.id, {sync: {status: 'waiting', updatedAt: Date.now()}})
+              this.walletInfos$.next(new Map(this.#walletInfos))
+            })
+            return from(wallets)
+          }),
+          concatMap((wallet) => {
+            this.#walletInfos.set(wallet.id, {sync: {status: 'syncing', updatedAt: Date.now()}})
+            this.walletInfos$.next(new Map(this.#walletInfos))
+            return from(wallet.sync()).pipe(
+              finalize(() => {
+                if (this.#walletInfos.get(wallet.id)?.sync.status !== 'error') {
+                  this.#walletInfos.set(wallet.id, {sync: {status: 'done', updatedAt: Date.now()}})
+                  this.walletInfos$.next(new Map(this.#walletInfos))
+                }
+              }),
+              catchError((error) => {
+                this.#walletInfos.set(wallet.id, {sync: {status: 'error', error, updatedAt: Date.now()}})
+                this.walletInfos$.next(new Map(this.#walletInfos))
+                return of()
+              }),
+            )
+          }),
+          finalize(() => {
+            this.isSyncing = false
+          }),
+        )
+        .subscribe()
+    }
+
+    const subscription = interval(thirtyFiveSeconds).subscribe(syncWallets)
+
+    return {
+      destroy: () => {
+        subscription.unsubscribe()
+      },
+    }
+  }
+
+  getOpenedWalletsByNetwork = () => {
```

Comment:
this is for Portfolio manager, we will have 1 store per network for the tokens, cached.